### DISCUSSION TITLE
MAUI Preview 4 does not work

### DISCUSSION THREAD

**Discussion by vonwenm** [dotnet/maui#1315] [Q&A]
I had been delighted the amazing maui was released for experimentation. ….and when it will be ready for even the most basic alpha testing in a later preview I will be delighted again.

Have been trying to get MAUI Preview 4 to work but there seems to be no way. Granted Maui-check is easy to deploy and works reliably. However then – when Maui-check is all good and green the whole thing comes apart and gets worse and worse with no way to fix.

If you try to create maui projects from the visual studio templates they break out of the box. any attempt to fix these breaks makes things worse. Preview 4 IS UNUSABLE. Please avoid the pain until this is fixed in a later PREVIEW.

Unless you like hundreds of error like the one below and no way of fixing them.

Error CS0234 The type or namespace name ‘AspNetCore’ does not exist in the namespace ‘Microsoft’ (are you missing an assembly reference?) MAUI_03.WinUI3 D:\MAUI\DEMO_03\MAUI_03\MAUI_03\MAUI_03.WinUI3\Microsoft.NET.Sdk.Razor.SourceGenerators\Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator_Pages_FetchData_razor.cs 193 Active

**Comment by vinit21** [comment]
Hello vonwenm,

I havent tried winUI but I got MAUI + Android  + iOS working after trying for 3 days.  Few points that helped me. 
1. use Visual Studio Preview. Keep it updated. 
2. You can get all check box checked in Maui-Check cmd.  It doesnt mean 100% maui app will run. 
3. For Android - run simulator with api 30 - x86_64 first before building. Then  Click Start ( Android Emultaor). Ony if your android emulator is running, VS will find it. Or else it will say "Device not found". 
4. It will successfully deploy the app but Visual Studio will stop debugging (i suspect it crashes when loading hot reload. But MAUI app is deployed on your android and you can run the app perfectly. 
4. iOS is also having similar issue. 

I hope it helps you.
